### PR TITLE
Listings close & unpublish

### PR DESCRIPTION
**Pull request by dominikx96**
# Pull Request Template

## Issue
Closes: #1443, #1442

Addresses # (issue)
- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description
Added `close` and `unpublish` buttons to the `<Aside />` component and modal to close a listing.

Current action buttons which are depending on the listing status:
- closed:
    - View listing
    - Save & exit
    - Publish
- pending:
    - View listing
    - Save & exit
    - Publish
- open:
    - View listing
    - Save & exit
    - Unpublished
    - Close

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes

### CODE REVIEW

File: backend/core/src/listings/listings.controller.ts [hunk 1]
```diff
@@ -133,4 +133,18 @@ export class ListingsController {
...
+    await this.listingsService.close(listingId)
+    await clearCacheKeys(this.cacheManager, [...this.cacheKeys, `/listings/${listingId}`])
+  }
+
+  @Put(`:listingId/unpublish`)
+  @ApiOperation({ summary: "Unpublish listing by id", operationId: "unpublish" })
+  async unpublish(@Param("listingId") listingId: string) {
+    await this.listingsService.unpublish(listingId)
+    await clearCacheKeys(this.cacheManager, [...this.cacheKeys, `/listings/${listingId}`])
+  }
```
Comment (seanmalbert) [inline backend/core/src/listings/listings.controller.ts:149]: @dominikx96 and @pbn4 , do we really want to maintain separate endpoints for closing, unpublishing, etc? If it does make sense to have endpoints to change the status, can we have one with a status param, like `@Put(':listingId/:status')`?

**Review by seanmalbert — CHANGES_REQUESTED** [review]
@dominikx96 ,

I think this looks great and works well. There's one minor styling issue compared to the comps where the close button is red. Aside from that there are two issues with this that I see:

1. I'm not sure we should have the "View Listing" button here, or at least not have it work as it does now. If a user has made a bunch of edits and then clicks "View Listing" they will lose all their work.
2. Similarly with "Close", the changes aren't saved, and though probably not as big of deal, a user may choose to make some updates before setting it to closed.

**Comment by dominikx96** [comment]
@seanmalbert 

I fixed both of the points from your list, agree that the "view listing" button doesn't have sense in this case. Also, fixed the problem with form data after the "close" action.

Not sure if I understand correctly, but I decided to keep the same color for the close buttons (in the aside and modal), because 'unpublish' is red so it makes a little confusion.

**Review by seanmalbert — APPROVED** [review]
Thanks for the updates, @dominikx96 . They look good.

**Pull request merged**